### PR TITLE
misc cleanup

### PR DESCRIPTION
**Pull request by andy-k**

### CODE REVIEW

File: liwords-ui/src/gameroom/table.tsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ export const Table = React.memo((props: Props) => {
 
   useEffect(() => {
     if (gameContext.playState === PlayState.GAME_OVER || isObserver) {
-      return () => {};
+      return;
```
Comment (domino14) [inline liwords-ui/src/gameroom/table.tsx:141]: my linter complained about not returning a function (consistent returns). Do we have the same linter settings? They should be checked into this repo.

File: liwords-ui/src/gameroom/table.tsx [hunk 2]
```diff
@@ -209,9 +201,7 @@ export const Table = React.memo((props: Props) => {
           }
         )
         .then((streakresp) => {
-          if (stillMountedRef.current) {
-            setStreakGameInfo(streakresp.data.game_info);
-          }
+          setStreakGameInfo(streakresp.data.game_info);
```
Comment (domino14) [inline liwords-ui/src/gameroom/table.tsx:204]: this one would probably need to be protected as it's an XHR - what do you think?

Comment (andy-k) [reply liwords-ui/src/gameroom/table.tsx:204]: it is. we're no longer using `React.useState` directly

**Review by domino14 — COMMENTED** [review]
as you mentioned, i am most worried about the anti-pattern of checking whether stuff is mounted. What is the alternative -- to cancel all AJAX requests in the right places?

**Comment by domino14** [comment]
It still seems worth merging as it's a clean way of setting state, and it seems to work, so i'll merge it.

**Pull request merged**